python/mpi_cmake_modules/clang_format.py

```diff
@@ -50,7 +50,25 @@ def test_valid_file(filename):
         return False
 
 
+# either returns file_or_directory as such (if exists)
```

Comment:
This should be a doc-string see the master branch I added the doc-string you should do a:
git rebase origin/master vberenz/relative_clang_format
git push origin vberenz/relative_clang_format -f

Reply:
this is a private function, no doc-string required.

Reply:
this is a private function, no doc-string required.

Reply:
ok, but it could be a none private function that would be used in the python formatter...

Reply:
???

Reply:
sanity check: by "none private" you mean public ?

Reply:
I meant without the "_" at the beginning of the function. Hence public yes.

Reply:
I do not see the point of making this function public.

Reply:
Because we are going to use it with the python code formatter whatever it is is. So that both use absolute path everywhere.

Reply:
Honestly I do not care so much, we should just merge what you have.

Reply:
what confuses me : I do not understand what you call "python code formatter". I thought this was the code of the python code formatter.

Reply:
ok, now I got what you mean. But for the python formatter, we will use smthg like black or autopep8 which will come with its own executable, no ?

Reply:
Yes but the principle will be the same:
- find the executable
- setup the parameters
- find the list of folders or file to apply it on (absolute path)
- execute the black or autopep8

So I expect to reuse most of the function of clang-format for the other formatter script.
And we will create a third formatter script wich will basycally execute both formatter on the given list.

WDYT?

Reply:
yop, but in this case the executable and the library should be separated. I'll do this and push again.

Reply:
I agree it would be clean, just remove the clang-format symbolik link in this case and replace it with and executable script.

Reply:
yep, but it was a confusing review: the original is an executable with most of the functions specific to clang (e.g hard coded extensions), but the tiny update I made had behave as if parts of a library general for other formaters.